### DISCUSSION TITLE
UID higher than 10000000

### DISCUSSION THREAD

**Discussion by ygmbleye** [containers/podman#21293] [General]
Hi,
Running rootless podman (newbie).
My UID is higher than 10000000, and subuid/subgid are setup. 
I get the following error: Error: lchown /containers/storage/volumes/0aee80f386959678b4958c30e7ff58e4911ee00c07bb/_data: invalid argument
Any way to workaround this?

Thanks

**Comment by eriksjolund** [comment]
Does it work with `--userns=keep-id:uid=$uid,gid=$gid` ?

I did a test on Fedora CoreOS 40.20231209.91.0 (rawhide)
```
$ id=10000001
$ podman run --rm --user $id:$id docker.io/library/alpine id
Error: OCI runtime error: crun: setgroups: Invalid argument
$ podman run --rm --user $id:$id --userns=keep-id:uid=$id,gid=$id docker.io/library/alpine id
uid=10000001(10000001) gid=10000001(10000001) groups=10000001(10000001)
$ podman --version
podman version 4.8.1
```

**Reply by ygmbleye** [reply]
Hi, 
yes, --userns=keep-id did the trick
Thanks!

**Comment by rhatdan** [comment]
@giuseppe is this a crun issue?

**Reply by giuseppe** [reply]
It is probably using a higher ID that is currently possible in the user namespace, so the error is expected in this case
